### ISSUE TITLE
LeanData.cs Hard-Coded Market.USA

### ISSUE THREAD

**Issue by sl624** [QuantConnect/Lean#1464]
```csharp

        public static Symbol ReadSymbolFromZipEntry(Symbol symbol, Resolution resolution, string zipEntryName)
        {
            //....
            var isHourlyOrDaily = resolution == Resolution.Hour || resolution == Resolution.Daily;
            var parts = zipEntryName.Replace(".csv", string.Empty).Split('_');
            switch (symbol.ID.SecurityType)
            {

                case SecurityType.Future:
                    if (isHourlyOrDaily)
                    {
                        var expiryYearMonth = DateTime.ParseExact(parts[2], DateFormat.YearMonth, null);
                        var futureExpiryFunc = FuturesExpiryFunctions.FuturesExpiryFunction(parts[1]);
                        var futureExpiry = futureExpiryFunc(expiryYearMonth);
                        return Symbol.CreateFuture(parts[0], Market.USA, futureExpiry);
                    }
                    else
                    {
                        var expiryYearMonth = DateTime.ParseExact(parts[4], DateFormat.YearMonth, null);
                        var futureExpiryFunc = FuturesExpiryFunctions.FuturesExpiryFunction(parts[1]);
                        var futureExpiry = futureExpiryFunc(expiryYearMonth);
                        return Symbol.CreateFuture(parts[1], Market.USA, futureExpiry);
                    }

```

should we change all Market.USA to symbol.ID.Market?

**Comment by AlexCatarino** [comment]
@sl624 , good catch. Do you mind making a PR to fix it?

**Comment by mchandschuh** [comment]
In two cases the market should come from `symbol.ID.Market` and a third case it should come from `config.Market`.

**Comment by jaredbroad** [comment]
Fixed in #1493
